hw3/exercise-3/src/components/Search.js

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from "react";
+//import {FaSearch} from "react-icons/fa"
+import "./search.css";
+
+const Search = () => {
+	const [character, setCharacter] = useState("");
+	const [selected, setSelected] = useState(null);
+
+	const url = "https://thronesapi.com/api/v2/Characters";
+	const fetchData = (value) => {
+		fetch(url)
+			.then((response) => {
+				return response.json();
+			})
+			.then((data) => {
+				const results = data.find((character) => {
+					return (
+						value &&
+						character.fullName.toLowerCase().includes(value.toLowerCase())
+					);
+				});
+				if (results) {
+					setSelected(results);
+				} else {
+					setSelected(null);
+				}
+			});
+	};
+
+	let inputHandler = (value) => {
+		setCharacter(value);
+		fetchData(value);
+	};
+
+	return (
+		<div>
+			<div className="searchBar">
+				<div className="user-input">
+					<input
+						placeholder="Type to search..."
+						value={character}
+						onChange={(e) => inputHandler(e.target.value)}
```

Comment:
With this you are requesting all 53 characters every single time you enter a character into the list. A much more efficient way to go about this would be to request the data on page load with a useEffect hook and keep it in memory.